DRAFT.md

```diff
@@ -740,15 +737,13 @@ constexpr indirect(allocator_arg_t, const Allocator& alloc,
 
 21. _Mandates_: `T` is a complete type.
 
-22. _Effects_: Equivalent to `indirect(allocator_arg, alloc, *other)`.
+22. _Effects_: TODO.
 
 ```c++
 constexpr indirect(indirect&& other) noexcept;
 ```
 
-23. _Effects_: Constructs an `indirect` that takes ownership of the `other`'s
-    owned object and stores the address in `p_`. `allocator_` is initialized by
-    construction from `other.allocator_`.
+23. _Effects_: Equivalent to `indirect(allocator_arg_t{}, Allocator(), std::move(other))`.
```

Comment:
Here is how unique_ptr handles specifying the effect via the post condition here: https://eel.is/c++draft/unique.ptr.single.ctor:

```
constexpr unique_ptr(unique_ptr&& u) noexcept;
[15] Constraints: is_move_constructible_v<D> is true
[16] Preconditions: If D is not a reference type, D meets the Cpp17MoveConstructible requirements (Construction of the deleter from an rvalue of type D does not throw an exception[.](https://eel.is/c++draft/unique.ptr.single.ctor#16.sentence-2)
[17] Effects: Constructs a unique_ptr from u. If D is a reference type, this deleter is copy constructed from u's deleter; otherwise, this deleter is move constructed from u's deleter 
[Note: The construction of the deleter can be implemented with std​::​forward<D> — end note]
[18] Postconditions: get() yields the value u.get() yielded before the construction u.get() == nullptr get_deleter() returns a reference to the stored deleter that was constructed from u.get_deleter() If D is a reference type then get_deleter() and u.get_deleter() both reference the same lvalue deleter
```

Reply:
Nice. I reckon that's half of the story. Allocators might complicate matters.